docs: comprehensive update of user guides (#2534)
ApeWorX/ape: docs/userguides/forking_networks.md

## Forked Context

If you are already connected to a live network and wish to temporarily fork it, use the [fork() context manager](../methoddocs/managers.html#ape.managers.networks.NetworkManager.fork):

```python
from ape import networks

with networks.ethereum.mainnet.use_provider("alchemy") as alchemy:
    # Connect to live network
    print(f"Connected to: {alchemy.name}")
    

>     # Create a fork of the current network

Reviewer: ```suggestion
    # Create a fork of the current network using the specified fork provider
```

    with networks.fork(provider_name="foundry") as fork:
        print(f"Now using fork: {fork.name}")
        

>     # You can specify a block number
>     with networks.fork(provider_name="foundry", block_number=17000000) as fork:

Reviewer: ```suggestion
    # You can specify a block number (using the configured default fork provider)
    with networks.fork(block_number=17000000) as fork:
```

        print(f"Using fork at block {fork.chain.blocks.height}")
```

## Time Travel in Forks

One powerful feature of forking is the ability to manipulate block timestamps for testing time-dependent contracts:

```python
from ape import networks, chain

> with networks.ethereum.mainnet.fork(provider_name="foundry") as provider:

Reviewer: ```suggestion
with networks.ethereum.mainnet.fork() as provider:
```

>     # Get current timestamp
>     current_time = chain.pending_timestamp
>     
>     # Advance time by 7 days
>     chain.pending_timestamp = current_time + 7 * 24 * 60 * 60

Reviewer: ```suggestion
    # Advance time by 7 days
    chain.pending_timestamp += 7 * 24 * 60 * 60
```
